azure-monitor-opentelemetry-distro/azure/monitor/opentelemetry/distro/configurator.py

```diff
@@ -4,8 +4,22 @@
 # license information.
 # --------------------------------------------------------------------------
 
+
+from azure.monitor.opentelemetry.distro.diagnostic_logger import (
+    AzureDiagnosticLogger,
+)
 from opentelemetry.sdk._configuration import _OTelSDKConfigurator
 
 
 class AzureMonitorConfigurator(_OTelSDKConfigurator):
-    pass
+    def _configure(self, **kwargs):
+        try:
+            super()._configure(**kwargs)
+        except ValueError as e:
+            AzureDiagnosticLogger.log_diagnostic_error(
```

Comment:
Is it our plan to add the functionality of turning diagnostic logging feature off?

Reply:
Currently, it is automatically on for App Service and off everywhere else. My sense was that it is required by App Service so I didn't consider turning it off. Do you think I should add an env var override for diagnostic and status loggers?

Reply:
This was based on a conversation we had before the break. But we could switch it back to automatically allow diagnostic and status logger everywhere and then allow users to disable it. We could also extend it after this pr. Let me know what you think

Reply:
Resolution: Have it default OFF, and add functions `is_diagnostic_logger_enabled()` which checks for if it's in appservice.